AutoRest/Generators/CSharp/CSharp/SyncMethodsGenerationMode.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+namespace Microsoft.Rest.Generator.CSharp
+{
+    /// <summary>
+    /// Defines supported modes for sync wrapper generation
+    /// </summary>
+    public enum SyncMethodsGenerationMode
+    {
+        Essential,
+        All,
```

Comment:
Please add comments do get the idea what is the difference between essential and all.